examples/text-generation/run_lm_eval_quat_and_msr.py

```diff
@@ -0,0 +1,199 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###############################################################################
+# Copyright (C) 2020-2021 Habana Labs, Ltd. an Intel Company
+###############################################################################
+
+import argparse
+import json
+import logging
+import os
+import time
+
+import lm_eval.evaluator
+import lm_eval.tasks
+import torch
+import torch.nn.functional as F
+from run_generation import setup_parser
+from utils import initialize_model, finalize_quantization, setup_quantization
+
+from optimum.habana.utils import get_hpu_memory_stats
+
+
+os.environ.setdefault("TOKENIZERS_PARALLELISM", "false")
+logger = logging.getLogger(__name__)
+
+
+def setup_lm_eval_parser():
```

Comment:
as mentioned in our talk, suggest to move all code that it common to lm_eval to some file (lm_eval_utils or something)

Reply:
done